zookeeper-server/src/test/java/org/apache/zookeeper/common/X509UtilTest.java

```diff
@@ -389,6 +393,27 @@ public void testGetSslHandshakeDetectionTimeoutMillisProperty() {
         }
     }
 
+    @Test
+    public void testCreateSSLContext_invalidCustomSSLContextClass() {
+        ZKConfig zkConfig = new ZKConfig();
+        ClientX509Util clientX509Util = new ClientX509Util();
+        zkConfig.setProperty(clientX509Util.getSslClientContextProperty(), String.class.getCanonicalName());
+        try {
+            clientX509Util.createSSLContext(zkConfig);
+            fail("SSLContextException expected.");
```

Comment:
nit: can remove the try-catch and modify the annotation on the method to be `@Test(expected = X509Exception.SSLContextException.class)`. Though this approach does not check the message, I think it's probably ok - testing for specific error messages is fragile anyway. We should either not care about the message, or we should create a specific sublcass of SSLContextException for this case and check for that exception type.

Reply:
+1 Use `@Test(expected = ...)` expectation instead.

Reply:
Yep, I'll have that changed.